File: mall-project/mall-frontend/src/api/address.ts
import request from './request'

export interface Address {
  id: number
  userId: number
  receiverName: string
  receiverPhone: string
  province: string
  city: string
  district: string
  detail: string
  isDefault: number
  fullAddress?: string
  createTime?: string
  updateTime?: string
}

export interface SaveAddressRequest {
  receiverName: string
  receiverPhone: string
  province: string
  city: string
  district: string
  detail: string
  isDefault?: number
}

export const getAddressList = () => request.get<Address[]>('/address/list')

export const getDefaultAddress = () => request.get<Address | null>('/address/default')

export const createAddress = (data: SaveAddressRequest) => request.post<number>('/address', data)

export const updateAddress = (id: number, data: SaveAddressRequest) => request.put(`/address/${id}`, data)

export const deleteAddress = (id: number) => request.delete(`/address/${id}`)

export const setDefaultAddress = (id: number) => request.put(`/address/${id}/default`)

File: mall-project/mall-backend/src/main/java/com/mall/entity/Address.java
package com.mall.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 收货地址实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("address")
public class Address extends BaseEntity {

    private Long userId;

    private String receiverName;

    private String receiverPhone;

    private String province;

    private String city;

    private String district;

    private String detail;

    /**
     * 1-默认 0-非默认
     */
    private Integer isDefault;
}


